### PR TITLE
Prepare for 7.6.0

### PR DESCRIPTION
**Pull request by simonvik**
Reviewers must check the following items:

- [x] Release notes are complete (major release only)
- [x] Release notes no longer target trunk (major release only)
- [x] The change log is populated
- [x] The VRT history in `include/vrt.h` is populated
- [x] The VRT history refers to the next VRT version
- [x] The macro `VRT_MAJOR_VERSION` was updated if applicable
- [x] The macro `VRT_MINOR_VERSION` was updated if applicable
- [x] The new VRT version follows releases guidelines
- [x] The new VRT version matches the one from the VRT history
- [ ] Bundled `devicedetect.vcl` was updated (major release only)
- [x] Running `./autogen.des && make distcheck` succeeds
- [x] The version in `configure.ac` is correct
- [x] The copyright notice in `configure.ac` covers the current year
- [x] The copyright output in `lib/libvarnish/version.c` covers the current year
- [x] There are no other changes than the ones listed above

### CODE REVIEW

File: doc/sphinx/whats-new/changes-7.6.rst [hunk 1]
```diff
@@ -1,11 +1,11 @@
-.. _whatsnew_changes_CURRENT:
+.. _whatsnew_changes_7.6:
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-Changes in Varnish **${NEXT_RELEASE}**
+Changes in Varnish 7.6
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Comment (dridi) [inline doc/sphinx/whats-new/changes-7.6.rst:5]: Maybe align surrounding `%` lines with the length of the title?

File: doc/sphinx/whats-new/changes-7.6.rst [hunk 2]
```diff
@@ -25,14 +25,14 @@ varnishd
 
 A new ``linux`` jail has been added (configured via the ``-j`` argument) which is
 now the default on Linux. For now, it is almost identical to the ``unix`` jail
-with one :ref:`whatsnew_upgrading_CURRENT_linux_jail` added.
+with one :ref:`whatsnew_upgrading_7.6_linux_jail` added.
 
 The port of a *listen_endpoint* given with the ``-a`` argument to ``varnishd``
 can now also be a numerical port range like ``80-89``, besides the existing
 options of port number (e.g. ``80``) and service name (e.g. ``http``). With a
 port range, Varnish will accept connections on all ports within the range.
```
Comment (dridi) [inline doc/sphinx/whats-new/changes-7.6.rst:33]: Is the range inclusive? I think it should tell here even if it is not the reference manual.

File: doc/sphinx/whats-new/index.rst [hunk 3]
```diff
@@ -13,7 +13,7 @@ This section describes the changes and improvements between different
 versions of Varnish, and what upgrading between the different versions
 entail.
 
-Varnish **$NEXT_RELEASE**
+Varnish 7.6
 -------------------------
```
Comment (dridi) [inline doc/sphinx/whats-new/index.rst:17]: Please align the `-` line below to the new title length.

File: doc/sphinx/whats-new/index.rst [hunk 4]
```diff
@@ -23,10 +23,10 @@ released versions of Varnish, see the chapters listed below.**
...
    :maxdepth: 2
 
-   changes-trunk
-   upgrading-trunk
+   changes-7.6
+   upgrading-7.6
 
-Varnish **7.5**
+Varnish 7.5
 ---------------
```
Comment (dridi) [inline doc/sphinx/whats-new/index.rst:30]: Please align the `-` line below to the new title length.

File: doc/sphinx/whats-new/upgrading-7.6.rst [hunk 5]
```diff
@@ -1,14 +1,10 @@
...
-updates for changes in the development branch. For changes in the
-released versions of Varnish, see:** :ref:`whats-new-index`
-
-.. _whatsnew_upgrading_CURRENT:
+.. _whatsnew_upgrading_7.6:
 
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
-Upgrading to Varnish **$NEXT_RELEASE**
+Upgrading to Varnish 7.6
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```
Comment (dridi) [inline doc/sphinx/whats-new/upgrading-7.6.rst:5]: Please align the surrounding `%` lines to the new title length.

File: doc/changes.rst [hunk 6]
```diff
@@ -34,9 +34,9 @@ http://varnish-cache.org/docs/trunk/whats-new/index.html and via
 individual releases. These documents are updated as part of the
 release process.
 
-===============================
-Varnish Cache NEXT (2024-09-15)
-===============================
+================================
+Varnish Cache 7.6.0 (2024-09-15)
+================================
```
Comment (dridi) [inline doc/changes.rst:39]: This was properly aligned, but the date looks wrong.

File: include/vrt.h [hunk 7]
```diff
@@ -57,7 +57,7 @@
  * Whenever something is deleted or changed in a way which is not
  * binary/load-time compatible, increment MAJOR version
  *
- * NEXT (2024-09-15)
+ * 20.0 (2024-09-13)
```
Comment (dridi) [inline include/vrt.h:60]: See previous comment, based on the date here.

**Review by dridi — CHANGES_REQUESTED** [review]
I will read the release notes and start checking boxes while you are busy with the following :]

File: doc/sphinx/whats-new/changes-7.6.rst (hunk 2 above)
Comment (simonvik) [reply doc/sphinx/whats-new/changes-7.6.rst:33]: for (i = lo; i <= hi && !error; i++) {

Its inclusive and thats the only thing that makes sense?

Comment (dridi) [reply doc/sphinx/whats-new/changes-7.6.rst:33]: Not sure it's the only one, I had to pause and wonder.

**Review by dridi — APPROVED** [review]
I'm not seeing a problem :)

**Pull request merged**